### PR TITLE
Add paging to monitor's scheduling page

### LINKED ISSUE #1645: Paging at Monitor's /scheduling page
The speed of jobs loading at /scheduling page decreases with higher amount of jobs. Previous attempt to make it better (#1501) actually introduced a bug (#1638).

Better alternative is to provide paging at this page.

### PR DESCRIPTION
**Pull request by arjansh**
Fixes #1645.

Added ScheduleDataPager, which is an extension of GWT's SimplePager used for adding paging functionality to DataCleaner monitor's scheduling page, which shows the deployed jobs.

Implemented an extensions of the FlowPanel class which implements GWT's HasRows and HasData interfaces to be able to use the ScheduleDataPager.

### CODE REVIEW

**Comment by jhorcicka** [comment]
I am already testing it. Actually, I've just finished and with unfortunate results. It seems all jobs are loaded at once and then it is just GUI separating them into pages. 

**jobs | avg time**
7 | (1.83 + 1.43 + .172) / 3
207 | (6.33 + 5.42 + 5.78) / 3
407 | (10.44 + 9.83 + 10.13) / 3
607 | (14.36 + 14.06 + 14.09) / 3

**Comment by arjansh** [comment]
@jhorcicka I'll give that a look.

**Comment by LosD** [comment]
@arjansh Can the pager be combined with the loader, so that it's the pager that initiates the actual loading?

**Comment by arjansh** [comment]
I use a ListDataProvider which provides the data for the Pager. This provider is loads all entries directly into memory. I can maybe use another DataProvider implementation, which loads the data when needed. I'll have to look a bit into that however, there is an AsyncDataProvider, which I guess does exactly what we want.

File: monitor/widgets/src/main/java/org/datacleaner/monitor/scheduling/widgets/SchedulingOverviewPanel.java [hunk 1]
```diff
@@ -74,15 +69,16 @@ public void onSuccess(final List<ScheduleDefinition> result) {
...
+                                    .get(categoryAndGroupMapForJobs.keySet().iterator().next()));
                 } else {
                     final DecoratedTabPanel tabPanel = new DecoratedTabPanel();
                     tabPanel.setWidth("100%");
                     Collection<String> jobCategories = categoryAndGroupMapForJobs.keySet();
                     jobCategories = sortJobCategories(jobCategories);
 
                     for (final String jobCategory : jobCategories) {
-                        final List<ScheduleDefinition> list = categoryAndGroupMapForJobs.get(jobCategory);
+                                final List<JobIdentifier> list = categoryAndGroupMapForJobs.get(jobCategory);
```
Comment (LosD) [inline monitor/widgets/src/main/java/org/datacleaner/monitor/scheduling/widgets/SchedulingOverviewPanel.java]: Indentation looks weird here.

File: monitor/widgets/src/main/java/org/datacleaner/monitor/scheduling/widgets/SchedulingOverviewPanel.java [hunk 2]
```diff
@@ -74,15 +69,16 @@ public void onSuccess(final List<ScheduleDefinition> result) {
                     panel.add(new Label("There are no jobs available."));
                     initWidget(panel);
                 } else if (categoryCount == 1) {
-                    getPagedPanel(result);
+                            getPagedPanel(categoryAndGroupMapForJobs
```
Comment (LosD) [inline monitor/widgets/src/main/java/org/datacleaner/monitor/scheduling/widgets/SchedulingOverviewPanel.java]: Indentation looks weird here

**Comment by LosD** [comment]
Oh wait, that was because I looked at a single commit in isolation. LGTM!

**Comment by LosD** [comment]
@jhorcicka Do you want to redo the performance testing?

**Comment by jhorcicka** [comment]
Yes, I did. It is much better now. 

jobs | avg time
207 | (2.85 + 2.73 + 2.59) / 3
407 | (3.58 + 3.85 + 3.71) / 3
607 | (5.66 + 5.45 + 5.06) / 3

**Comment by jhorcicka** [comment]
Even better now. The first time I experienced weird behaviour when adding more jobs made the performance faster. :-) Second time it seems to be "normal"&good. 

**jobs | avg time**
7 | (3.93 + 3.09 + 3.16) / 3
207 | (4.51 + 2.66 + 2.69) / 3
407 | (2.73 + 2.66 + 1.59) / 3
607 | (2.24 + 2.44 + 3.18) / 3
807 | (3.10 + 2.82 + 3.84) / 3

**Comment by jhorcicka** [comment]
can-not-go-to-end

It seems that the paginator does not allow to use the last button (go to the end).

**Comment by jhorcicka** [comment]
Tested and LGTM.

**Pull request merged**